src/compiler/checker.ts

```diff
@@ -10063,7 +10063,19 @@ namespace ts {
 
         // The outer type parameters are those defined by enclosing generic classes, methods, or functions.
         function getOuterTypeParametersOfClassOrInterface(symbol: Symbol): TypeParameter[] | undefined {

```

Comment:
note: this probably should be renamed now, right?

Reply:
My guess is that the correct thing is to instead call `getOuterTypeParameters` at some point, but I'm not sure where.

Reply:
> My guess is that the correct thing is to instead call getOuterTypeParameters from lookupTypeParameterNodes

If I understand correctly, you are proposing to exchange the implemented code path:
`lookupTypeParameterNodes -> getTypeParametersOfClassOrInterface -> getOuterTypeParametersOfClassOrInterface -> getOuterTypeParameters`
to
`lookupTypeParameterNodes -> getOuterTypeParameters`.

The difference would be that:
1. we don't call into `getTypeParametersOfClassOrInterface` to eventually reach `getOuterTypeParameters`
2. we wouldn't be calling `getLocalTypeParametersOfClassOrInterfaceOrTypeAlias`

So ye, I definitely agree that, with the current implementation, calling to `getLocalTypeParametersOfClassOrInterfaceOrTypeAlias` is:
- either redundant
- broken too

I've checked if I can (should?) obtain local type params of the same symbol and based on some existing behavior I think I should:
```ts
function create<T>(t: T) {
  return class<T2> {
    someMethod() {}
  };
}

// (method) (Anonymous class)<"outer", "local">.someMethod(): void
new (create("outer" as const))<'local'>()./**/someMethod();
```

We can see here that the local type param of this anonymous class is being printed. This **doesn't** happen with the current state of this PR for those inner expando~ functions.

So given that a local type params should also be obtained here and printed I would have to both obtain inner and outer type params in the `lookupTypeParameterNodes` for this case. Which is already **exactly** what the current code path should be doing (it's just not handling expando functions). 

So I wonder - do we really want to special case this situation here in the `lookupTypeParameterNodes`? Wouldn't it be easier to just introduce the required logic to `getOuterTypeParametersOfClassOrInterface`, `getLocalTypeParametersOfClassOrInterfaceOrTypeAlias` **and** rename them?

A sideways observation is that in the given example with the anonymous class (and in the tests that I've added to this PR) the outer type params are "appended" to the symbol that doesn't reference them at all. We can notice this: `(Anonymous class)<"outer", "local">` but that anonymous class has only a single type param. I somewhat find this confusing but I assume that this is how it was designed to work.

Reply:
@sandersn friendly 🏓